### PR TITLE
Server WFS3 update to latest (final) core specs

### PR DESCRIPTION
**Pull request by elpaso**
- extent -> spatial -> array of bboxes
- collections name -> id

See: https://github.com/qgis/QGIS/pull/32262#issuecomment-542630718

### CODE REVIEW

File: tests/testdata/qgis_server/api/test_wfs3_collections_project.json [hunk 1]
```diff
@@ -45,16 +49,21 @@ Content-Type: application/json
...
-        "spatial": [
-          [
-            8.203459307036344,
-            44.90139483904469,
-            8.203546993993488,
-            44.901482526001836
-          ]
-        ]
+        "spatial": {
+          "bbox": [
```
Comment (rouault) [inline tests/testdata/qgis_server/api/test_wfs3_collections_project.json]: there's an extra level of array here. Should be `"bbox": [ [ 8.203459307036344, 44.90139483904469, 8.203546993993488,44.901482526001836 ]]`

Comment (elpaso) [reply tests/testdata/qgis_server/api/test_wfs3_collections_project.json]: Fixed, thank you for the review!

**Review by rouault — APPROVED** [review]

**Comment by nyalldawson** [comment]
For the same reason as #32271, this MUST be fixed before release

**Review by pblottiere — APPROVED** [review]

**Comment by haubourg** [comment]
Kudos guys!

**Pull request merged**